### PR TITLE
Space Lua: Improve numeric types semantics

### PR DESCRIPTION
**Pull request by mjf**
Add numeric result normalization for arithmetic operations:

* Integer operations preserve int type, collapse -0 to +0
* Float operations preserve float type, maintain -0.0
* Division always produces float results

Fix for-loop variable type tracking:

* Loop variable type determined by start/step parameters
* Integer loops produce integer variables
* Float/mixed loops produce float-tagged variables

Implement table key normalization:

* Integer-valued floats (1.0, 2.0) normalize to integer keys
* All zero variants (0, 0.0, -0.0) map to same key
* Non-integer floats (1.5) remain distinct keys

Add tests for:

* Arithmetic type preservation across operators
* For-loop type modes (int/float/mixed)
* Table key equivalence and normalization
* Edge cases with zero, -0.0, and special values
* Standard library `type` and `math.type`

### CODE REVIEW

File: client/space_lua/eval.ts [hunk 1]
```diff
@@ -855,52 +920,52 @@ const operatorsMetaMethods: Record<string, {
...
-      return r;
+      return normalizeArithmeticResult(ax + bx, bothInt);
     },
   },
   "-": {
     metaMethod: "__sub",
     nativeImplementation: (a, b, _ctx, _sf, hints) => {
-      const { ax, bx, bothInt } = coerceNumericPair(a, b, hints);
-      const r = ax - bx;
+      const { ax, bx, bothInt, aZeroKind: _aZeroKind, bZeroKind } =
```
Comment (zefhemel) [inline client/space_lua/eval.ts]: FYI: I think you can just leave out any keys you don't care about (so `aZeroKind` in this case)

File: client/space_lua/eval.ts [hunk 2]
```diff
@@ -1775,21 +1843,74 @@ export function evalStatement(
...
+            const r = evalStatement(fr.block, localEnv, sf, returnOnReturn);
+            const res = isPromise(r) ? await r : r;
+            if (res !== undefined) {
+              if (isBreakSignal(res)) {
+                break;
+              }
+              return res;
+            }
+          }
+        } else {
```
Comment (zefhemel) [inline client/space_lua/eval.ts]: The `if` and `else` bodies are 99% the same (except for the second `for` condition is that right?), can this not be merged to reduce code duplication?

`std0 > 0 ? i  <= e0 : i >= e0` or something similar?

File: client/space_lua/eval.ts [hunk 3]
```diff
@@ -1799,50 +1920,82 @@ export function evalStatement(
...
               }
-              return runAsync(i + step, end, step);
-            });
-          } else if (r !== undefined) {
-            if (isBreakSignal(r)) {
-              break;
+              return r;
+            }
+          }
+        } else {
```
Comment (zefhemel) [inline client/space_lua/eval.ts]: Same story on code duplication as above?

File: client/space_lua/eval.ts (hunk 1 above)
Comment (mjf) [reply client/space_lua/eval.ts]: Removed/fixed.

File: client/space_lua/eval.ts (hunk 2 above)
Comment (mjf) [reply client/space_lua/eval.ts]: Done.

File: client/space_lua/eval.ts (hunk 3 above)
Comment (mjf) [reply client/space_lua/eval.ts]: Yeah. Deduplicated.
